### PR TITLE
Prev/Next Buttons

### PR DESCRIPTION
**Pull request by andrewadcock**
On page load, scroll to top and remove focus from the selected button.

Addresses #577

### CODE REVIEW

File: frontend/react/src/components/layout/FormNavigation.js [hunk 1]
```diff
@@ -13,13 +13,20 @@ class FormNavigation extends Component {
   render() {
     const { history, location, sections } = this.props;
 
+    console.log("items-sections", sections);
```
Comment (jgillis1) [inline frontend/react/src/components/layout/FormNavigation.js:16]: Remove console.log

File: frontend/react/src/components/layout/FormNavigation.js [hunk 2]
```diff
@@ -13,13 +13,20 @@ class FormNavigation extends Component {
...
+      if (section.subsections.length < 2) {
+        items.push(idToUrl(section.id))
+      } else {
+        section.subsections.forEach(subsection => {
+          items.push(idToUrl(subsection.id));
+        });
+      }
     });
 
+    console.log("items", items);
```
Comment (jgillis1) [inline frontend/react/src/components/layout/FormNavigation.js:28]: Remove console.log

**Review by jgillis1 — CHANGES_REQUESTED** [review]

File: frontend/react/src/components/layout/FormNavigation.js (hunk 1 above)
Comment (andrewadcock) [reply frontend/react/src/components/layout/FormNavigation.js]: Removed

File: frontend/react/src/components/layout/FormNavigation.js (hunk 2 above)
Comment (andrewadcock) [reply frontend/react/src/components/layout/FormNavigation.js]: Removed

**Review by mblevins-collabralink — APPROVED** [review]
Navigation appears to work correctly.

**Review by jgillis1 — APPROVED** [review]

**Comment by jgillis1** [comment]
QA Review has been completed and the PR is ready for 18F review.

**Review by cl-adavyd — APPROVED** [review]

File: frontend/react/src/components/Utils/ScrollToTop.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect, Fragment } from 'react';
+import { withRouter } from 'react-router-dom';
+
+function ScrollToTop({ history, children }) {
```
Comment (mgwalker) [inline frontend/react/src/components/Utils/ScrollToTop.js:4]:
```suggestion
function ScrollToTop({ history }) {
```

File: frontend/react/src/components/Utils/ScrollToTop.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            // Remove focus from clicked button
+            document.activeElement.blur()
+
+        });
+        return () => {
+            unlisten();
+        }
+    }, []);
+
+    return <Fragment>{children}</Fragment>;
```
Comment (mgwalker) [inline frontend/react/src/components/Utils/ScrollToTop.js:18]:
```suggestion
    return null;
```

File: frontend/react/src/reactRouter.js [hunk 5]
```diff
@@ -40,35 +41,37 @@ const Routes = ({ userData }) => (
...
-            path="/sections/:year/:sectionOrdinal/:subsectionMarker"
-            children={<InvokeSection userData={userData} />}
-          />
-          <Route
-            path="/sections/:year/:sectionOrdinal"
-            children={<InvokeSection userData={userData} />}
-          />
-          <Route exact path="/test" component={test} />
-        </Switch>
+        <ScrollToTop>
```
Comment (mgwalker) [inline frontend/react/src/reactRouter.js:44]:
```suggestion
        <ScrollToTop />
```

File: frontend/react/src/reactRouter.js [hunk 6]
```diff
@@ -40,35 +41,37 @@ const Routes = ({ userData }) => (
...
+              path="/sections/:year/:sectionOrdinal/:subsectionMarker"
+              children={<InvokeSection userData={userData} />}
+            />
+            <Route
+              path="/sections/:year/:sectionOrdinal"
+              children={<InvokeSection userData={userData} />}
+            />
+            <Route exact path="/test" component={test} />
+          </Switch>
+        </ScrollToTop>
```
Comment (mgwalker) [inline frontend/react/src/reactRouter.js:74]:
```suggestion
```

**Review by mgwalker — APPROVED** [review]
I love this approach.

I just have a recommendation below that tweaks the `<ScrollToTop>` component so that it doesn't wrap the rest of the app. The main thing is that is shortens the render tree and makes things a tiny bit easier to find in the dev tools. I'm going to approve this because it's fine as-is, but I'm going to leave it unmerged in case you want to make the suggested changes. In either case, feel free to merge it whenever you're ready.

File: frontend/react/src/components/Utils/ScrollToTop.js (hunk 3 above)
Comment (andrewadcock) [reply frontend/react/src/components/Utils/ScrollToTop.js]: Super smart and super clean!

**Comment by andrewadcock** [comment]
This appears to have stopped working correctly. Related: The vertical nav is displaying duplicates of sections. More investigation is needed.

**Comment by andrewadcock** [comment]
After a reboot, the nav is working again. Unsure of the issue but it appears to be resolved.

**Review by jgillis1 — APPROVED** [review]

**Pull request merged**